test/common/run-tests

```diff
@@ -356,8 +358,9 @@ def detect_tests(test_files, image, opts):
                 rwa = not testlib.get_decorator(test_method, test, "no_retry_when_changed")
                 todo = testlib.get_decorator(test_method, test, "todo")
                 if getattr(test.__class__, "provision", None):
-                    # each additionally provisioned VM costs destructive test capacity
-                    cost = len(test.__class__.provision)
+                    # each additionally provisioned VM memory costs destructive test capacity
+                    total_memory = sum(config.get('memory_mb', 0) for _, config in test.__class__.provision.items())
+                    cost = math.ceil(total_memory / DEFAULT_MACHINE_MEMORY_SIZE)
```

Comment:
This doesn't work correctly now, as the intention for destructive/multi-VM tests is to have 1152 MiB, not 1400. That neeeds to be fixed in bots. Once that constant gets fixed to 1152 in bots, it is correct.